### ISSUE TITLE
Database Projects - Generate script not working for new Azure SQL DBs

### ISSUE THREAD

**Issue by zijchen** [microsoft/azuredatastudio#20133]
- Azure Data Studio Version: RC build from https://mssqltools.visualstudio.com/CrossPlatBuildScripts/_build/results?buildId=163875&view=results
- OS Version: Win 11

Steps to Reproduce:

1. Create SDK style project, add some tables/views/etc.
2. Build and generate script to a new database, the script task says it succeeded but the output script is empty

![image](https://user-images.githubusercontent.com/13544267/180106912-3eeace5c-4ed8-46f9-9e66-3890518cde09.png)

Curious if it is related to #20082

Does this issue occur when all extensions are disabled?: Yes/No

**Comment by ssreerama** [comment]
weird, I tried the same thing but I can see the script generated. I installed the RC from the build and installed SQL DB v0.18.0., created an SDK style proj and added a table and view. (In Win10)..

![image](https://user-images.githubusercontent.com/74571829/180115865-cfbcaccc-67bb-4deb-9a04-5d0c69b99167.png)

**Comment by kisantia** [comment]
I'm also not able to repro this on the latest RC build with sql database projects 0.18.0. Let's check again if this still happens after the fix for #20082 gets into the release branch

**Comment by kisantia** [comment]
Update: this is only happening on Azure when generating script for a database that doesn't exist and the script doesn't contain any of the expected create table and view statements, just this:
```
PRINT N'Creating database $(DatabaseName)...'
GO
CREATE DATABASE [$(DatabaseName)] COLLATE SQL_Latin1_General_CP1_CI_AS
GO
DECLARE  @job_state INT = 0;
DECLARE  @index INT = 0;
DECLARE @EscapedDBNameLiteral sysname = N'$(DatabaseName)'
WAITFOR DELAY '00:00:30';
WHILE (@index < 60) 
BEGIN
	SET @job_state = ISNULL( (SELECT SUM (result)  FROM (
		SELECT TOP 1 [state] AS result
		FROM sys.dm_operation_status WHERE resource_type = 0 
		AND operation = 'CREATE DATABASE' AND major_resource_id = @EscapedDBNameLiteral AND [state] = 2
		ORDER BY start_time DESC
		) r), -1);

	SET @index = @index + 1;

	IF @job_state = 0 /* pending */ OR @job_state = 1 /* in progress */ OR @job_state = -1 /* job not found */ OR (SELECT [state] FROM sys.databases WHERE name = @EscapedDBNameLiteral) <> 0
		WAITFOR DELAY '00:00:30';
	ELSE 
    	BREAK;
END
GO
```

When generating script for an Azure db that already exists or to an on prem db that doesn't exist, the script gets created with the expected create table and view statements. I've checked back to the April release and found this was happening since at least then.

Sqlpackage generates the script as expected, as well as calling the `DacServices.Script()` api in a small test program Zi made, so have to dig into more why this is having different behavior.
